src/hooks/useSessionWithWebApp.ts

```diff
@@ -0,0 +1,165 @@
+type IWebAppParameterDefinition = IShapeDiverParameterDefinition;
+
+/** Type used for export definitions */
+type IWebAppExportDefinition = IShapeDiverExportDefinition;
+
+/** Reference to a parameter (custom or defined by the session) */
+interface IWebAppParameterRef {
+	/** Optional id of the referenced parameter. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced parameter. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the parameter to be overridden. */
+	overrides?: Pick<IWebAppParameterDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+	/** Disable the UI element of the parameter if its state is dirty. */
+	disableIfDirty: boolean
+	/** Ask the user to accept or reject changes of this parameter before executing them. */
+	acceptRejectMode: boolean
+}
+
+/** Reference to an export (defined by the session) */
+interface IWebAppExportRef {
+	/** Optional id of the referenced export. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced export. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the export to be overridden. */
+	overrides?: Pick<IWebAppExportDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+}
+
+/** Types of widgets */
+type WebAppWidgetType = "accordion" | "text" | "image";
+
+/** 
+ * Properties of a parameter and export accordion widget.
+ * UI elements of the referenced parameters and exports are grouped 
+ * and ordered according to their properties (which might be overridden).
+ */
+interface IWebAppWidgetPropsAccordion {
+	/** References to parameters which shall be displayed by the accordion. */
+	parameters: IWebAppParameterRef[]
+	/** References to exports which shall be displayed by the accordion. */
+	exports: IWebAppExportRef[]
+	/** 
+	 * Optional name of group that should be used for all parameters/exports without a group.
+	 * In case this is not specified, parameters/exports without a group will be displayed without an accordion.
+	 */
+	defaultGroupName?: string
+}
+
+/** Properties of a text widget. */
+interface IWebAppWidgetPropsText {
+	/** Plain text. Takes precedence. */
+	text?: string
+	/** Optional markdown. */
+	markdown?: string
+}
+
+/** Properties of an image widget. */
+interface IWebAppWidgetPropsImage {
+	/** Base 64 encoded image. Takes precedence over export reference. */
+	data?: string
+	/** Optional reference to export which provides the image */
+	export?: IWebAppExportRef
+}
+
+/** A widget. */
+interface IWebAppWidget {
+	/** Type of the widget. */
+	type: WebAppWidgetType
+	/** Properties of the widget. */
+	props: IWebAppWidgetPropsAccordion | IWebAppWidgetPropsText | IWebAppWidgetPropsImage
+}
+
+/** 
+ * Types of icons for tabs 
+ * @see https://tabler.io/icons
+ */
+type WebAppContainerTabIconType = "replace" | "download";
```

Comment:
@mathieu-shapediver please add icons that you think we should support out of the box.